agent/engine/docker_task_engine.go

```diff
@@ -936,44 +945,34 @@ func (engine *DockerTaskEngine) isParallelPullCompatible() bool {
 	return false
 }
 
-// GetAdditionalAttributes returns back attributes of the instance when register
-// the instance into cluster
-func (engine *DockerTaskEngine) GetAdditionalAttributes() []*ecs.Attribute {
-	var attributes []*ecs.Attribute
-
-	attribute, ok := engine.taskNetworkAttributes()
-	if !ok {
+// getTaskENIAttribute checks if the task network was enabled and whether the plugin are existed
+// TODO This method needs to be refactored to extract both version and feature flags from the
+// plugins. The task-eni capability will become a name-only capability and the version
+// information will be registered as an instance attribute
+func (engine *DockerTaskEngine) getTaskENIAttribute() *ecs.Attribute {
+	if !engine.cfg.TaskENIEnabled {
 		return nil
 	}
 
-	for key, value := range attribute {
-		attributes = append(attributes, &ecs.Attribute{
-			Name:  aws.String(key),
-			Value: aws.String(value),
-		})
-	}
-
-	return attributes
-}
-
-// taskNetworkAttributes checks if the task network was enabled and whether the plugin are existed
-func (engine *DockerTaskEngine) taskNetworkAttributes() (map[string]string, bool) {
+	// ECS Agent requires all of these plugins to configure the ENI for a task
 	plugins := []string{"ecs-bridge", "ecs-eni", "ecs-ipam"}
 
-	if engine.cfg.TaskENIEnabled {
-		// Check if all the plugin existed in the specific directory
-		for _, plugin := range plugins {
-			_, err := engine.cniClient.Version(plugin)
-			if err != nil {
-				log.Error("Engine: Check version of plugin %s failed", plugin)
-				return nil, false
-			}
+	for _, plugin := range plugins {
+		// Check if we can get version information from each plugin
+		version, err := engine.cniClient.Version(plugin)
+		if err != nil {
```

Comment:
`cmd.Output` should be capturing all of that anyway IIUC https://golang.org/src/os/exec/exec.go?s=12344:12382#L452